core/cont/inc/THashList.h

```diff
@@ -36,7 +36,7 @@ class THashTable;
 class THashList : public TList {
 
 protected:
-   THashTable   *fTable;    //Hashtable used for quick lookup of objects
+   THashTable   *fTable;    //! Hashtable used for quick lookup of objects
```

Comment:
Dear Philippe, 

> Optimistically one would hope (s)he would have also added some I/O test ... Also likely the existing test trying to read existing object of that type (assuming there is any such test ...) would start to fail but not silently, they ought to fail with issues about byte-count.

Yes, my usual approach is not too optimistic (maybe I saw too much framework code from the experimental site recently - the people I work with usually ignore test results nor do they write tests :-( ). That's also my main motivation for the tool: To have some basic tests generic enough to test some functionality of almost all classes. 
Would there really be a byte-count warning, though? The realdata layout is effectively unchanged by the new member since the class version is 0. 



> I think we can help this by replacing (well superseding) the 0 by a constexpr unsigned short named kAllDataMembersTransient (okay that's too long of a name :( ) valued at 0 for backward compatibility.

That's a cool idea! I sadly do not have a good idea for a shorter name (the name is indeed long, but expressive...), but I love it. It's easy to check programmatically, and also expressive to the reader of the class - and fully backwards-compatible. 
For fresh developments I'd still recommend the "version 1 + `//!` " approach especially to beginners to clarify the IO intent "closer" to the member in the code, though. 

So I propose the following: 
- I remove the last commit of the commit series in this PR (i.e. the commit touching `TSeqCollection` / `THashList`). I actually did that just right now. 
- I add a suppression for these two classes to my tool (right now these seem to be the only cases of correct class-version-0 usage in IO-baseclasses I have encountered in the ROOT codebase - but I don't have full coverage yet, many of ROOTs more exotic, seldomly used classes segfault after default-construct/destruct, or default-construct + streaming mostly due to use of globals... The tool is right now filled with some ugly blacklist patterns to ignore some class-groups). 
- You could add something like `kAllDataMembersTransient` (maybe you indeed find a shorter name...) any time you want, you know much better than me in which header / namespace this constexpr should live. I'll see your commit in the commitlog and then extend my tool to check for that. 

Thanks a lot for the fruitful discussion, and the cool idea, as usual! 

Cheers, 
Oliver